internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,548 @@
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"align_checkpoints": {
+						Description: "Specifies whether the transform checkpoint ranges should be optimized for performance. Default value is true.",
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     true,
+					},
+					"dates_as_epoch_millis": {
+						Description: "Defines if dates in the output should be written as ISO formatted string (default) or as millis since epoch.",
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+					},
+					"deduce_mappings": {
+						Description: "Specifies whether the transform should deduce the destination index mappings from the transform config. The default value is true",
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     true,
+					},
+					"docs_per_second": {
+						Description: "Specifies a limit on the number of input documents per second. Default value is null, which disables throttling.",
+						Type:        schema.TypeFloat,
+						Optional:    true,
+					},
+					"max_page_search_size": {
+						Description: "Defines the initial page size to use for the composite aggregation for each checkpoint. The default value is 500.",
+						Type:        schema.TypeInt,
+						Optional:    true,
+					},
+					"num_failure_retries": {
+						Description: "Defines the number of retries on a recoverable failure before the transform task is marked as failed. The default value is the cluster-level setting num_transform_failure_retries.",
+						Type:        schema.TypeInt,
+						Optional:    true,
+					},
+					"unattended": {
+						Description: "In unattended mode, the transform retries indefinitely in case of an error which means the transform never fails. Defaults to false.",
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+					},
+				},
+			},
+		},
+		"defer_validation": {
+			Type:        schema.TypeBool,
+			Description: "When true, deferrable validations are not run upon creation, but rather when the transform is started. This behavior may be desired if the source index does not exist until after the transform is created.",
+			Optional:    true,
+			Default:     false,
+		},
+		"timeout": {
+			Type:         schema.TypeString,
+			Description:  "Period to wait for a response from Elastisearch when performing any management operation. If no response is received before the timeout expires, the operation fails and returns an error. Defaults to `30s`.",
+			Optional:     true,
+			Default:      "30s",
+			ValidateFunc: utils.StringIsDuration,
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Description: "Controls wether the transform is started or stopped. Default is `false` (stopped).",
+			Optional:    true,
+			Default:     false,
+		},
+	}
+
+	utils.AddConnectionSchema(transformSchema)
```

Comment:
```suggestion
```

The resource level connection schema has been deprecated, no need to add it to new resources.